File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015VB/VirtualRadGridViewSortingFiltering.Core/VirtualIndex.vb

Imports System.Collections.Generic
Imports System.Text
Imports Telerik.Collections.Generic
Imports Telerik.WinControls.Data
Imports Telerik.WinControls.UI

Namespace VirtualRadGridViewSortingFiltering.Core
    Public Class VirtualIndex
        Inherits Index(Of GridViewRowInfo)
        Public Sub New(collectionView As RadCollectionView(Of GridViewRowInfo))
            MyBase.New(collectionView)
        End Sub

        Public Overrides ReadOnly Property Items() As IList(Of GridViewRowInfo)
            Get
                Return TryCast(Me.CollectionView, VirtualGridDataView).ListSource
            End Get
        End Property

        Protected Overrides Sub Perform()
        End Sub
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015CS/VirtualRadGridViewSortingFiltering.Core/VirtualIndex.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.Collections.Generic;
using Telerik.WinControls.Data;
using Telerik.WinControls.UI;

namespace VirtualRadGridViewSortingFiltering.Core
{
    public class VirtualIndex : Index<GridViewRowInfo>
    {
        public VirtualIndex(RadCollectionView<GridViewRowInfo> collectionView)
            : base(collectionView)
        {
        }

        public override IList<GridViewRowInfo> Items
        {
            get
            {
                return (this.CollectionView as VirtualGridDataView).ListSource;
            }
        }

        protected override void Perform()
        {
        }
    }
}
